cs_core/15112.md

```diff
@@ -0,0 +1,106 @@
+- **Term project:** About the last month ish or so. Since you pick the idea, the
+standards needed to reach the MVP (grade = 80%) will vary, but they will make it 
+similar between years and similar projects types. You'll meet with your TA Mentor
+and submit project checkpoints each week, and also demo the final product. The
+demo = explain all the cool things/details you did so they'll give you all the 
+points you deserve.
+- **Final:** I'm sorry, don't look at me for this one. Unfortunately I don't see
+any previous final exams posted on the website, but they'll probably give you 
+practice problems.
+- **Hackathon:** Totally optional, but lots of fun, prizes, and beginner-friendly.
+You also get an extension on the hw if you attend (extension doesn't count towards 
+grace days, so you can use a grace day on top of your extension).
+
+## How to do well
+
+- Make sure you have all required technology up to standards. Installation instructions
+on website.
+- Do the bonuses on HW. (They're also on quizzes/tests but there's more time pressure.)
+- Do lots of practice for quizzes and tests (they also have extra coding practice
+if you want that).
+- The TA's hold optional solution sessions for the hw and review sessions for tests.
+They'll go over concepts, practice problems and answer questions.
+- Ask questions on Piazza/OH: the teachers like to brag that the response rate is
+<5 min. Piazza is good for specific questions, but if you want more general help 
+or are stuck with debugging, go to OH.
+- CITE YOUR CODE (comment where you got it from) or else you will lose lots of
+points and be sad. You should only be citing your own work (previous hw) or
+lecture/recitation. You're not allowed to look stuff up online, and they have a 
+plagarism detector.
+- Follow the coding style guidelines exactly (ex: when they say functions must 
+be <= 30 lines, that includes the function header. And no going over by even a 
+single line).
+- Start thinking about your term project ideas now. You can also meet with them 
+to discuss it, so they can give you feedback about whether it's viable.
+- You have 2 grace days (limit 1 per hw) where you can submit hw up to <24hr 
+late for full credit. So use them wisely if you need to (they can't be used for 
+term project deadlines though).
+- Complete the hw in time for OH help.
+- Don't rely on a curve. Sure they might curve a quiz a little bit in the beginning
+when people are still adjusting to the course, but it's definitely not a regular thing.
+- Term project counts for a good chunk of your grade so spend quality time on it.
+
+## What to watch out for
+
+- The first quiz, as mentioned above, is usually a bit rough.
+- HW gradually get harder over time, but the difficulty will flatten out by 
+week 4 according to the teachers. IMO I think the difficulty level depends on how
+much experience you've had with the topic before.
+- Technology failures: back up your code somewhere, and don't submit for the
```

Comment:
Is `git` a reasonable recommendation here for file management, or is that too complicated for 15-112? 

When I was at CMU, CS majors had to learn `git` in some class (although you barely did), but 112 is taken by so many non-CS majors, it's unlikely that most people will know what it is. 

Version control is extremely important at higher level classes, but getting a headstart in 112 could be a good idea